### PR TITLE
Allow intents as list in from_intent in slot mapping

### PR DESCRIPTION
**Pull request by Tawakalt**
**Proposed changes**:
- Fixes [ATO-919](https://rasahq.atlassian.net/browse/ATO-919)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

[ATO-919]: https://rasahq.atlassian.net/browse/ATO-919?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ancalita — COMMENTED** [review]
Hey @Tawakalt I will review this now, but thought of flagging that since this is a regression bug, it might be better to target `3.5.x` (all bugfixes should only be merged to release branches).

**Comment by Tawakalt** [comment]
> Hey @Tawakalt I will review this now, but thought of flagging that since this is a regression bug, it might be better to target `3.5.x` (all bugfixes should only be merged to release branches).

@ancalita Oh okay! Even though it was introduced in the last release?
I wasn't aware of the last clause 👍🏾

File: rasa/shared/core/slot_mappings.py [hunk 1]
```diff
@@ -205,14 +205,15 @@ def check_mapping_validity(
...
             and mapping.get(INTENT) is not None
-            and mapping.get(INTENT) not in domain.intents
         ):
-            rasa.shared.utils.io.raise_warning(
-                f"Slot '{slot_name}' uses a 'from_intent' mapping for "
-                f"a non-existent intent '{mapping.get('intent')}'. "
-                f"Skipping slot extraction because of invalid mapping."
-            )
-            return False
+            for intent in mapping.get(INTENT):
```
[the remark this answers is not in the document]
Comment (Tawakalt) [reply rasa/shared/core/slot_mappings.py:209]: Oh yeah! I assumed it was always a list. I'll adjust it.

File: tests/shared/core/test_slot_mappings.py [hunk 2]
```diff
@@ -167,3 +167,79 @@ def test_slot_mappings_check_mapping_validity_from_intent():
...
+
+        slots:
+          {slot_name}:
+            type: any
+            influence_conversation: false
+            mappings:
+            - type: from_intent
+              value: "testing 123"
+              intent:
+                - affirm
```
[the remark this answers is not in the document]
Comment (Tawakalt) [reply tests/shared/core/test_slot_mappings.py:229]: Yeah, I was testing for the content of the list instead.

[the remark this answers is not in the document]
Comment (ancalita) [reply tests/shared/core/test_slot_mappings.py:229]: Oh I see, with intents that are not listed in the Domain, this is indeed a good test case.

[the remark this answers is not in the document]
Comment (Tawakalt) [reply tests/shared/core/test_slot_mappings.py:229]: yes

**Comment by Tawakalt** [comment]
closing this to target 3.5

**Pull request closed without merging**